hypervisor/arch/x86/guest/instr_emul.c

```diff
@@ -374,14 +373,12 @@ static uint64_t vm_get_register(const struct acrn_vcpu *vcpu, enum cpu_reg_name
  * @pre ((reg <= CPU_REG_LAST) && (reg >= CPU_REG_FIRST))
  * @pre ((reg != CPU_REG_CR2) && (reg != CPU_REG_IDTR) && (reg != CPU_REG_GDTR))
  */
-static void vm_set_register(struct acrn_vcpu *vcpu, enum cpu_reg_name reg,
-								uint64_t val)
+static void vm_set_register(struct acrn_vcpu *vcpu, enum cpu_reg_name reg, uint64_t val)
 {
 
-	if ((reg >= CPU_REG_GENERAL_FIRST) && (reg <= CPU_REG_GENERAL_LAST)) {
+	if (reg >= CPU_REG_GENERAL_FIRST) {
```

Comment:
should be if (reg <= CPU_REG_GENERAL_LAST) { ?

Reply:
Sorry, my fault